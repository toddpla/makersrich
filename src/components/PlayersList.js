import React, { Component } from 'react';
import database from '../firebase/firebase'
import PlayersListItem from './PlayersListItem'


class PlayersList extends Component {

  render() {
    return (
      <table>
        <tr>
          <th></th><th>Player</th><th>Cash</th><th>Rubies</th><th>Beans</th>
        </tr>
        {this.props.players.map((player, i) => <PlayersListItem {...player}/>)}
      </table>
    );
  }
}

export default PlayersList;
