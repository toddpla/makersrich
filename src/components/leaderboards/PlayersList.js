import React, { Component } from 'react';
import PlayersListItem from './PlayersListItem'

export class PlayersList extends Component {


  render() {
    return (
      <table id="leaderboard-table">
      <tbody>
        <tr className="leaderboard-table-headings">
          <th></th><th>Player</th><th>Cash</th><th>Rubies</th><th>Beans</th>
        </tr>
        {this.props.players.map((player, i) => <PlayersListItem {...player}/>)}
        </tbody>
      </table>
    );
  }
}

export default PlayersList;
