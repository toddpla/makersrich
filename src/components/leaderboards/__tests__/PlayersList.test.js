import React from 'react'
import { shallow } from 'enzyme'
import playersData from '../../../test/fixtures/players'
import { PlayersList } from '../PlayersList'

let players, wrapper;

beforeEach(() => {
  players = playersData.fixtures
  wrapper = shallow(<PlayersList players={players} />)
})

test('should render correctly', () => {
  expect(wrapper).toMatchSnapshot()
})
