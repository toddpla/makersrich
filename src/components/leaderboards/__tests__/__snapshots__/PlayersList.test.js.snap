// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayersList
    players={
      Array [
        Object {
          "cash": 0,
          "displayName": "Frankie",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "1",
        },
        Object {
          "cash": 0,
          "displayName": "Stanley",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "2",
        },
        Object {
          "cash": 0,
          "displayName": "Anna",
          "left": 0,
          "level": 1,
          "state": "online",
          "top": 0,
          "uid": "3",
        },
        Object {
          "cash": 0,
          "displayName": "Jenny",
          "left": 0,
          "level": 1,
          "state": "offline",
          "top": 0,
          "uid": "4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <tbody>
        <tr
          className="leaderboard-table-headings"
        >
          <th />
          <th>
            Player
          </th>
          <th>
            Cash
          </th>
          <th>
            Rubies
          </th>
          <th>
            Beans
          </th>
        </tr>
        <PlayersListItem
          cash={0}
          displayName="Frankie"
          left={0}
          level={1}
          state="online"
          top={0}
          uid="1"
        />
        <PlayersListItem
          cash={0}
          displayName="Stanley"
          left={0}
          level={1}
          state="online"
          top={0}
          uid="2"
        />
        <PlayersListItem
          cash={0}
          displayName="Anna"
          left={0}
          level={1}
          state="online"
          top={0}
          uid="3"
        />
        <PlayersListItem
          cash={0}
          displayName="Jenny"
          left={0}
          level={1}
          state="offline"
          top={0}
          uid="4"
        />
      </tbody>,
      "id": "leaderboard-table",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <tr
            className="leaderboard-table-headings"
          >
            <th />
            <th>
              Player
            </th>
            <th>
              Cash
            </th>
            <th>
              Rubies
            </th>
            <th>
              Beans
            </th>
          </tr>,
          Array [
            <PlayersListItem
              cash={0}
              displayName="Frankie"
              left={0}
              level={1}
              state="online"
              top={0}
              uid="1"
            />,
            <PlayersListItem
              cash={0}
              displayName="Stanley"
              left={0}
              level={1}
              state="online"
              top={0}
              uid="2"
            />,
            <PlayersListItem
              cash={0}
              displayName="Anna"
              left={0}
              level={1}
              state="online"
              top={0}
              uid="3"
            />,
            <PlayersListItem
              cash={0}
              displayName="Jenny"
              left={0}
              level={1}
              state="offline"
              top={0}
              uid="4"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th />,
              <th>
                Player
              </th>,
              <th>
                Cash
              </th>,
              <th>
                Rubies
              </th>,
              <th>
                Beans
              </th>,
            ],
            "className": "leaderboard-table-headings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Player",
              },
              "ref": null,
              "rendered": "Player",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cash",
              },
              "ref": null,
              "rendered": "Cash",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rubies",
              },
              "ref": null,
              "rendered": "Rubies",
              "type": "th",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Beans",
              },
              "ref": null,
              "rendered": "Beans",
              "type": "th",
            },
          ],
          "type": "tr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cash": 0,
            "displayName": "Frankie",
            "left": 0,
            "level": 1,
            "state": "online",
            "top": 0,
            "uid": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cash": 0,
            "displayName": "Stanley",
            "left": 0,
            "level": 1,
            "state": "online",
            "top": 0,
            "uid": "2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cash": 0,
            "displayName": "Anna",
            "left": 0,
            "level": 1,
            "state": "online",
            "top": 0,
            "uid": "3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cash": 0,
            "displayName": "Jenny",
            "left": 0,
            "level": 1,
            "state": "offline",
            "top": 0,
            "uid": "4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "tbody",
    },
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <tr
            className="leaderboard-table-headings"
          >
            <th />
            <th>
              Player
            </th>
            <th>
              Cash
            </th>
            <th>
              Rubies
            </th>
            <th>
              Beans
            </th>
          </tr>
          <PlayersListItem
            cash={0}
            displayName="Frankie"
            left={0}
            level={1}
            state="online"
            top={0}
            uid="1"
          />
          <PlayersListItem
            cash={0}
            displayName="Stanley"
            left={0}
            level={1}
            state="online"
            top={0}
            uid="2"
          />
          <PlayersListItem
            cash={0}
            displayName="Anna"
            left={0}
            level={1}
            state="online"
            top={0}
            uid="3"
          />
          <PlayersListItem
            cash={0}
            displayName="Jenny"
            left={0}
            level={1}
            state="offline"
            top={0}
            uid="4"
          />
        </tbody>,
        "id": "leaderboard-table",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr
              className="leaderboard-table-headings"
            >
              <th />
              <th>
                Player
              </th>
              <th>
                Cash
              </th>
              <th>
                Rubies
              </th>
              <th>
                Beans
              </th>
            </tr>,
            Array [
              <PlayersListItem
                cash={0}
                displayName="Frankie"
                left={0}
                level={1}
                state="online"
                top={0}
                uid="1"
              />,
              <PlayersListItem
                cash={0}
                displayName="Stanley"
                left={0}
                level={1}
                state="online"
                top={0}
                uid="2"
              />,
              <PlayersListItem
                cash={0}
                displayName="Anna"
                left={0}
                level={1}
                state="online"
                top={0}
                uid="3"
              />,
              <PlayersListItem
                cash={0}
                displayName="Jenny"
                left={0}
                level={1}
                state="offline"
                top={0}
                uid="4"
              />,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th />,
                <th>
                  Player
                </th>,
                <th>
                  Cash
                </th>,
                <th>
                  Rubies
                </th>,
                <th>
                  Beans
                </th>,
              ],
              "className": "leaderboard-table-headings",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Player",
                },
                "ref": null,
                "rendered": "Player",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cash",
                },
                "ref": null,
                "rendered": "Cash",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Rubies",
                },
                "ref": null,
                "rendered": "Rubies",
                "type": "th",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Beans",
                },
                "ref": null,
                "rendered": "Beans",
                "type": "th",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cash": 0,
              "displayName": "Frankie",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "1",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cash": 0,
              "displayName": "Stanley",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "2",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cash": 0,
              "displayName": "Anna",
              "left": 0,
              "level": 1,
              "state": "online",
              "top": 0,
              "uid": "3",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "cash": 0,
              "displayName": "Jenny",
              "left": 0,
              "level": 1,
              "state": "offline",
              "top": 0,
              "uid": "4",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
